actors/builtin/storage_market/storage_market_actor.go

```diff
@@ -38,24 +34,27 @@ func (a *StorageMarketActor) WithdrawBalance(rt Runtime, params *WithdrawBalance
 		rt.Abort(exitcode.ErrIllegalArgument, "negative amount %v", params.Amount)
 	}
 
-	recipientAddr := builtin.RT_MinerEntry_ValidateCaller_DetermineFundsLocation(rt, params.Address, builtin.MinerEntrySpec_MinerOrSignable)
+	recipientAddr := builtin.MarketAddress(rt, params.Address)
 
 	var amountExtracted abi.TokenAmount
 	var st StorageMarketActorState
 	rt.State().Transaction(&st, func() interface{} {
-		st._rtAbortIfAddressEntryDoesNotExist(rt, params.Address)
+		st.abortIfAddressEntryDoesNotExist(rt, params.Address)
 
 		// Before any operations that check the balance tables for funds, execute all deferred
 		// deal state updates.
 		//
 		// Note: as an optimization, implementations may cache efficient data structures indicating
 		// which of the following set of updates are redundant and can be skipped.
-		amountSlashedTotal = big.Add(amountSlashedTotal, st._rtUpdatePendingDealStatesForParty(rt, params.Address))
+		amountSlashedTotal = big.Add(amountSlashedTotal, st.updatePendingDealStatesForParty(rt, params.Address))
+
+		minBalance := st.getLockedBalance(rt, params.Address)
 
-		minBalance := st._getLockedReqBalanceInternal(params.Address)
-		newTable, ex, ok := autil.BalanceTable_WithExtractPartial(st.EscrowTable, params.Address, params.Amount, minBalance)
-		Assert(ok)
-		st.EscrowTable = newTable
+		et := adt.AsBalanceTable(adt.AsStore(rt), st.EscrowTable)
+		ex, err := et.SubtractWithMinimum(params.Address, params.Amount, minBalance)
+		AssertNoError(err)
```

Comment:
Be very careful with assert, I think we should not have many of them (including this one). This is legacy from un-executable spec code. The right thing here is to rt.Abort().